docs/source/data/advanced_data_catalog_usage.md

```diff
@@ -0,0 +1,258 @@
+# Advanced Data Catalog usage
+
+## How to read the same file using two different datasets
+
+Use transcoding to load and save a file via its specified `filepath` using more than one `DataSet` implementation.
+
+### A typical example of transcoding
+
+Parquet files can not only be loaded via the `ParquetDataSet` using `pandas`, but also directly by `SparkDataSet`. This conversion is typical when coordinating a `Spark` to `pandas` workflow.
+
+To enable transcoding, define two `DataCatalog` entries for the same dataset in a common format (Parquet, JSON, CSV, etc.) in your `conf/base/catalog.yml`:
+
+```yaml
+my_dataframe@spark:
+  type: spark.SparkDataSet
+  filepath: data/02_intermediate/data.parquet
+  file_format: parquet
+
+my_dataframe@pandas:
+  type: pandas.ParquetDataSet
+  filepath: data/02_intermediate/data.parquet
+```
+
+These entries are used in the pipeline like this:
+
+```python
+pipeline(
+    [
+        node(func=my_func1, inputs="spark_input", outputs="my_dataframe@spark"),
+        node(func=my_func2, inputs="my_dataframe@pandas", outputs="pipeline_output"),
+    ]
+)
+```
+
+### How does transcoding work?
+
+In this example, Kedro understands that `my_dataframe` is the same dataset in its `spark.SparkDataSet` and `pandas.ParquetDataSet` formats and helps resolve the node execution order.
+
+In the pipeline, Kedro uses the `spark.SparkDataSet` implementation for saving and `pandas.ParquetDataSet`
+for loading, so the first node outputs a `pyspark.sql.DataFrame`, while the second node receives a `pandas.Dataframe`.
+
+
+## How to access the Data Catalog in code
```

Comment:
I found this transition unnatural, from transcoding to using Data Catalog in Code. Or maybe this page is just a list of advance distinct use cases?

Reply:
I've reflected and have moved this into the examples section and reduced the size so it's no longer such a weird confusing thing, just a simple example. I doubt anyone is using it much but it is there just in case.